tracpro/orgs_ext/middleware.py

```diff
@@ -0,0 +1,32 @@
+from django.utils.translation import ugettext_lazy as _
+from django.http import HttpResponseBadRequest
+
+from temba_client.base import TembaAPIError, TembaConnectionError
+
+
+class HandleTembaAPIError(object):
+    """ Catch all Temba exception errors """
+
+    def process_exception(self, request, exception):
+
+        rapidProConnectionErrorString = _(
+            "RapidPro appears to be down right now. "
+            "Please try again later.")
+
+        if isinstance(exception, TembaAPIError):
+            rapidpro_connection_error_codes = ["502", "503", "504"]
+
+            if any(code in exception.caused_by.message for code in rapidpro_connection_error_codes):
```

Comment:
On a side note with that, I propose that we only check for 5XX/TembaConnectionErrors and 403 errors. Everything else is probably specific to the view, either an error in our code :frowning: or something like a client data error that needs to be handled by validation in the view.